### PR TITLE
configure strict pytest

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
a remake of https://github.com/encode/httpcore/pull/364

**Comment by tomchristie** [comment]
Thanks @graingert!

**Comment by tomchristie** [comment]
Hrrrmm... I think the warnings that I'm seeing here may be unclosed resources from the `httpbin` fixture that we're using, rather than actually from `httpcore`.

(I guess I could be wrong, but I'm seeing the proper socket closes when digging through the codebase.)

**Comment by tomchristie** [comment]
Related - Ideally I'd rather replace our usage of a package based fixture here, with a plain ol' bit of socket code.
Just prefer a more slimline profile, rather than "neat bit of magic that spins up a whole httpbin server". I mean that was great to get going with, but I'd probably rather see a more down-to-basics, "a socket running in another thread that'll always just issue a 200 OK Hello World response".

File: setup.cfg [hunk 1]
```diff
@@ -19,9 +19,12 @@ known_third_party = brotli,certifi,chardet,cryptography,h11,h2,hstspreload,pytes
 skip = httpcore/_sync/,tests/_sync
 
 [tool:pytest]
-addopts = -rxXs
+addopts = -rxXs --strict-config --strict-markers
 markers =
   copied_from(source, changes=None): mark test as copied from somewhere else, along with a description of changes made to accodomate e.g. our test setup
+filterwarnings =
+    error
+    ignore:unclosed <ssl.SSLSocket .*:ResourceWarning
```
Comment (graingert) [inline setup.cfg:27]:
```suggestion
    ignore:unclosed <(socket\.socket|ssl\.SSLSocket) .*:ResourceWarning
```

File: setup.cfg [hunk 2]
```diff
@@ -19,9 +19,12 @@ known_third_party = brotli,certifi,chardet,cryptography,h11,h2,hstspreload,pytes
 skip = httpcore/_sync/,tests/_sync
 
 [tool:pytest]
-addopts = -rxXs
+addopts = -rxXs --strict-config --strict-markers
 markers =
   copied_from(source, changes=None): mark test as copied from somewhere else, along with a description of changes made to accodomate e.g. our test setup
+filterwarnings =
+    error
+    ignore:unclosed <(socket\.socket|ssl\.SSLSocket) .*:ResourceWarning
```
Comment (graingert) [inline setup.cfg:27]:
```suggestion
    ignore:unclosed <(socket\.socket|ssl\.SSLSocket) .*:ResourceWarning
    ignore:ssl\.wrap_socket\(\) is deprecated, use SSLContext\.wrap_socket\(\):DeprecationWarning
```

File: setup.cfg [hunk 3]
```diff
@@ -19,9 +19,13 @@ known_third_party = brotli,certifi,chardet,cryptography,h11,h2,hstspreload,pytes
...
+    ignore:ssl\.wrap_socket\(\) is deprecated, use SSLContext\.wrap_socket\(\):DeprecationWarning
```
Comment (graingert) [inline setup.cfg:28]:
```suggestion
    ignore:ssl\.wrap_socket\(\) is deprecated, use SSLContext\.wrap_socket\(\):DeprecationWarning
    ignore:ssl.PROTOCOL_TLS is deprecated:DeprecationWarning
```

File: setup.cfg [hunk 4]
```diff
@@ -19,9 +19,14 @@ known_third_party = brotli,certifi,chardet,cryptography,h11,h2,hstspreload,pytes
...
+    ignore:ssl\.PROTOCOL_TLS is deprecated:DeprecationWarning
```
Comment (graingert) [inline setup.cfg:29]: https://github.com/kevin1024/pytest-httpbin/pull/65/files

**Review by tomchristie — APPROVED** [review]
Good work on getting this to a resolution. Thanks!

**Pull request merged**